src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import cn from 'classnames';
+
+import { FilterType, Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[],
+  filterType: FilterType,
+  setFilterType: (filter: FilterType) => void,
+  handleClearCompleted: () => void,
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  todos,
+  filterType,
+  setFilterType,
+  handleClearCompleted,
+}) => {
+  const itemsLeft = todos.filter((todo) => !todo.completed).length;
+
+  return (
+    <footer className="todoapp__footer">
+      <span className="todo-count">
+        {`${itemsLeft} items left`}
+      </span>
+
+      <nav className="filter">
+        <a
+          href="#/"
+          className={cn(
+            'filter__link',
+            {
+              selected: filterType === FilterType.ALL,
+            },
+          )}
+          onClick={() => setFilterType(FilterType.ALL)}
```

Comment:
Create separate handler instead of using anonymous function. Fix all similar cases